lib/index.js

```diff
@@ -213,7 +213,9 @@ Segment.prototype.normalize = function(msg) {
       ctx.traits.crossDomainId = crossDomainId;
     }
   }
-  if (query) ctx.campaign = utm(query);
+  if (query && !ctx.campaign) {
+    ctx.campaign = utm(query);
```

Comment:
I started to do that initially, but then thought that campaign properties usually go together as a group, so there is some risk of inadvertently merging campaign properties from two different groups (if overrides and UTM differ).

happy to do it either way and I can make the change quickly if desired.